CONTRIBUTING.md

```diff
@@ -43,6 +43,19 @@ querying network status and applying network states.
 
 - `./tests/` - Contains tests for unit and integration tests.
 
+## Installation and Compilation:
+
+- [Fork](https://github.com/nmstate/nmstate.git) and git clone to local
+-[Install rust and Cargo for the development of nmstate](https://doc.rust-lang.org/cargo/getting-started/installation.html) or `apt-get install cargo`
+- Change your current directory to the nmstate directory `cd nmstate`
+- Install nmstate `cargo install nmstatectl`
+This will download the necessary dependencies and build the nmstate library and command-line tool
+- Build and compile the packages using `cargo build`
+This will compile the source code and generate the executable files
```

Comment:
Change executable for binary. Some files are libraries, not executables, for example.